plonky2x/src/backend/circuit/serialization.rs

```diff
@@ -200,6 +201,24 @@ impl<F: RichField + Extendable<D>, const D: usize> WitnessGeneratorRegistry<F, D
     pub fn register_simple<SG: SimpleGenerator<F, D>>(&mut self, id: String) {
         self.register::<SimpleGeneratorAdapter<F, SG, D>>(id)
     }
+
+    pub fn register_hint(&mut self, hint_fn: fn(&mut ValueStream<F, D>, &mut ValueStream<F, D>)) {
```

Comment:
Should we have ReadableValueStream and WriteableValueStream? Right now it looks like you could read from the second stream.

Reply:
I agree with this

Reply:
I named them ValueReader and ValueWriter